### PR TITLE
Microsoft Teams - improved error handling in long running loop

### PR DESCRIPTION
**Pull request by Itay4**
## Status
- [x] Ready

## Description
Improved error handling to handle errors such as seen in this log:

```
2020-05-11 04:03:35.7157 info (teamsitay_Microsoft Teams_long-running-execution) Started channel mirror loop thread (source: /home/circleci/.go_workspace/src/github.com/demisto/server/services/automation/dockercoderunner.go:716) 
2020-05-11 04:03:35.7166 error (teamsitay_Microsoft Teams_long-running-execution) An error occurred in long running loop: Expecting value: line 1 column 1 (char 0) (source: /home/circleci/.go_workspace/src/github.com/demisto/server/services/automation/dockercoderunner.go:718) 
2020-05-11 04:03:35.7173 error (teamsitay_Microsoft Teams_long-running-execution) An error occurred in channel mirror loop: Expecting ',' delimiter: line 1 column 8195 (char 8194) (source: /home/circleci/.go_workspace/src/github.com/demisto/server/services/automation/dockercoderunner.go:718) 
2020-05-11 04:03:35.7177 info (teamsitay_Microsoft Teams_long-running-execution) Full Integration Log:
Command being called is long-running-execution
Expecting value: line 1 column 1 (char 0) - Traceback (most recent call last):
  File "<string>", line 5323, in long_running_loop
  File "<string>", line 129, in info
  File "<string>", line 161, in __do
  File "/usr/local/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 5389, in main
  File "<string>", line 5352, in long_running_loop
ValueError: Expecting value: line 1 column 1 (char 0)
 (source: /home/circleci/.go_workspace/src/github.com/demisto/server/services/automation/dockercoderunner.go:716) 
```

`json.loads()` in `channel_mirror_loop` raised a `json.decoder.JSONDecodeError` which led to an exception raised in `long_running_loop` which crashed the thread

Added sleep to retry and start the loop again 

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 1]
```diff
@@ -1571,54 +1579,60 @@ def long_running_loop():
...
+            error_message = str(e)
+        finally:
+            if certificate_path:
+                os.unlink(certificate_path)
+            if private_key_path:
+                os.unlink(private_key_path)
+            if error_message:
+                demisto.error(f'An error occurred in long running loop: {error_message}')
+                demisto.updateModuleHealth(f'An error occurred: {error_message}')
+            time.sleep(5)
```
Comment (liorblob) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1635]: do we want to loop again after we unlinked all the certificates?

Comment (Itay4) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1635]: they will be created again in the next iteration, so i think its ok

Comment (liorblob) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1635]: The files may be already exist no?

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 2]
```diff
@@ -1571,54 +1579,60 @@ def long_running_loop():
...
+            server = WSGIServer(('0.0.0.0', port), APP, **ssl_args)
+            server.serve_forever()
+        except Exception as e:
+            error_message = str(e)
+        finally:
+            if certificate_path:
+                os.unlink(certificate_path)
+            if private_key_path:
+                os.unlink(private_key_path)
+            if error_message:
```
Comment (glicht) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1632]: For clarity you can put this code in the Exception handler and not the finally.

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 3]
```diff
@@ -1571,54 +1579,60 @@ def long_running_loop():
...
-            os.unlink(certificate_path)
-        if private_key_path:
-            os.unlink(private_key_path)
-        demisto.error(f'An error occurred in long running loop: {str(e)}')
-        raise ValueError(str(e))
+            server = WSGIServer(('0.0.0.0', port), APP, **ssl_args)
+            server.serve_forever()
+        except Exception as e:
+            error_message = str(e)
+        finally:
```
Comment (glicht) [inline Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1627]: I think in the finally you want to stop the server.  So you probably want to add something of the sort:
```
if server:
    server.stop()
```

File: Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py (hunk 1 above)
Comment (Itay4) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1635]: not sure i follow, can you give an example ?

Comment (liorblob) [reply Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py:1634]: never mind, it's a temporary file

**Comment by Itay4** [comment]
will open new PR

**Pull request closed without merging**